resource/docker/network/preparer.go

```diff
@@ -0,0 +1,163 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network
+
+import (
+	"errors"
+
+	"github.com/asteris-llc/converge/load/registry"
+	"github.com/asteris-llc/converge/resource"
+	"github.com/asteris-llc/converge/resource/docker"
+	dc "github.com/fsouza/go-dockerclient"
+	"golang.org/x/net/context"
+)
+
+// Preparer for docker networks
+//
+// Network is responsible for managing Docker networks. It assumes that there is
+// already a Docker daemon running on the system.
+type Preparer struct {
+	// name of the network
+	Name string `hcl:"name" required:"true"`
+
+	// network driver. default: bridge
+	Driver string `hcl:"driver"`
+
+	// labels to set on the network
+	Labels map[string]string `hcl:"labels"`
+
+	// driver specific options
+	Options map[string]interface{} `hcl:"options"`
+
+	// ip address management driver
+	IPAMDriver string `hcl:"ipam_driver"`
+
+	// custom IPAM configuration. multiple IPAM configurations are permitted. Each
+	// IPAM configuration block should contain one or more of the following items:
+	//
+	//   * subnet:      subnet in CIDR format
+	//   * gateway:     ipv4 or ipv6 gateway for the corresponding subnet
+	//   * ip_range:    container ips are allocated from this sub-ranges (CIDR format)
+	//   * aux_address: auxiliary ipv4 or ipv6 addresses used by the network driver.
+	//                  Aux addresses are specified as a map with a name key and an IP
+	//                  address value
+	IPAMConfig []ipamConfigMap `hcl:"ipam_config"`
+
+	// restricts external access to the network
+	Internal bool `hcl:"internal"`
+
+	// enable ipv6 networking
+	IPv6 bool `hcl:"ipv6"`
+
+	// indicates whether the volume should exist.
+	State State `hcl:"state" valid_values:"present,absent"`
+
+	// indicates whether or not the volume will be recreated if the state is not
+	// what is expected. By default, the module will only check to see if the
+	// volume exists. Specified as a boolean value
+	Force bool `hcl:"force"`
+}
+
+// Prepare a docker network
+func (p *Preparer) Prepare(ctx context.Context, render resource.Renderer) (resource.Task, error) {
+	if p.Name == "" {
```

Comment:
I think this is needed to prevent someone from manually specifying `name = ""` in the hcl